bots/pr/src/main/java/org/openjdk/skara/bots/pr/PullRequestCheckIssueVisitor.java

```diff
@@ -236,7 +242,7 @@ public void visit(WhitespaceIssue e) {
     @Override
     public void visit(MessageIssue issue) {
         var message = String.join("\n", issue.commit().message());
-        throw new IllegalStateException("Incorrectly formatted commit message: " + message);
+        throw new BadCommitMessageException("Incorrectly formatted commit message: " + message);
```

Comment:
Looking more at this, communicating through an exception seems wrong. The visitor already has a way to communicate problems through the addFailureMessage method. We should just stop throwing exception and treat this as any other "normal" failure.

Reply:
> Looking more at this, communicating through an exception seems wrong. The visitor already has a way to communicate problems through the addFailureMessage method. We should just stop throwing exception and treat this as any other "normal" failure.

Yes, this way is better. I was just trying to make bad commit message as an `integrationBlocker`. If we choose the way you mentioned, we also no longer need the `BadCommitMessageException`, right?